### PR TITLE
Updated linux installer to fix issues with 5.1

### LINKED ISSUE #230: `vboxadd-service` not being started under VirtualBox 5.1; causing errors
As of VirtualBox 5.1, there are two init scripts provided by the guest additions, `/etc/init.d/vboxadd` and `/etc/init.d/vboxadd-service`; however, vbguest only starts `vboxadd`. It appears that both are necessary for, e.g., IP address detection. This causes a hang for a minute or so after `==> default: Configuring and enabling network interfaces...` and then the following error to be printed:

```
No guest IP was given to the Vagrant core NFS helper. This is an
internal error that should be reported as a bug.
```

The guest additions configure `vboxadd-service` to start on boot (on RHEL/CentOS, by running `chkconfig vboxadd-service start` as part of setup), so doing a `vagrant reload` will typically fix things.

I confirmed that if I ssh into a box while it's hung at `Configuring and enabling network interfaces...` and run `sudo /etc/init.d/vboxadd-service start`, it'll come up fine.

I'm not actually sure if this is a vbguest problem or a VirtualBox problem, since it seems like `vboxadd.sh` should just start the service. But it also seems reasonable for vbguest to check that both `vboxadd` and `vboxadd-service` had been started.

Full output from our environment:

```
% vagrant up --provider=virtualbox
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Importing base box 'devvm'...
==> default: Matching MAC address for NAT networking...
==> default: Checking if box 'devvm' is up to date...
==> default: Setting the name of the VM: ep-vm-devvm_default_1470961282583_55866
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default:
    default: Vagrant insecure key detected. Vagrant will automatically replace
    default: this with a newly generated keypair for better security.
    default:
    default: Inserting generated public key within guest...
    default: Removing insecure key from the guest if it's present...
    default: Key inserted! Disconnecting and reconnecting using new SSH key...
==> default: Machine booted and ready!
[default] No installation found.
Loaded plugins: fastestmirror, priorities, protectbase, remove-with-leaves,
              : security, versionlock
Setting up Install Process
Loading mirror speeds from cached hostfile
 * base: mirrors.cat.pdx.edu
 * epel: mirror.prgmr.com
 * extras: linux.mirrors.es.net
 * updates: mirror.tzulo.com
1050 packages excluded due to repository priority protections
0 packages excluded due to repository protections
No package kernel-devel-4.7.0-1.el6.elrepo.x86_64 available.
Package gcc-4.4.7-17.el6.x86_64 already installed and latest version
Package binutils-2.20.51.0.2-5.44.el6.x86_64 already installed and latest version
Package 1:make-3.81-23.el6.x86_64 already installed and latest version
Package 4:perl-5.10.1-141.el6_7.1.x86_64 already installed and latest version
Package bzip2-1.0.5-7.el6_0.x86_64 already installed and latest version
Nothing to do
Copy iso file /Applications/VirtualBox.app/Contents/MacOS/VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
mount: /tmp/VBoxGuestAdditions.iso is write-protected, mounting read-only
Installing Virtualbox Guest Additions 5.1.2 - guest version is unknown
Verifying archive integrity... All good.
Uncompressing VirtualBox 5.1.2 Guest Additions for Linux...........
VirtualBox Guest Additions installer
Copying additional installer modules ...
Installing additional modules ...
vboxadd.sh: Building Guest Additions kernel modules.
vboxadd.sh: Starting the VirtualBox Guest Additions.

Could not find the X.Org or XFree86 Window System, skipping.
==> default: Checking for guest additions in VM...
==> default: Setting hostname...
==> default: Configuring and enabling network interfaces...
No guest IP was given to the Vagrant core NFS helper. This is an
internal error that should be reported as a bug
```

### PR DESCRIPTION
**Pull request by cdloh**
Since VirtualBox 5.1 there is an extra service that needs starting.

I've updated the install code so that it is a bit more flexible and starts the services straight after install.

This fixes #230

### CODE REVIEW

File: lib/vagrant-vbguest/installers/linux.rb [hunk 1]
```diff
@@ -166,21 +175,21 @@ def systemd_tool
...
-      # Checks for the correct location of the 'vboxadd' tool.
+      # Checks for the correct location of the tool provided.
       # It checks for a given list of possible locations. This list got
       # extracted from the 'VBoxLinuxAdditions.run' script.
       #
-      # @return [String|nil] Absolute path to the +vboxadd+ tool,
+      # @return [String|nil] Absolute path to the tool,
       #                      or +nil+ if none found.
-      def vboxadd_tool
+      def find_tool(tool)
```
Comment (bitbauer) [inline lib/vagrant-vbguest/installers/linux.rb]: Method are referenced by vboxadd_tool() anytime! This seems to be a lose end.

Comment (Roguelazer) [inline lib/vagrant-vbguest/installers/linux.rb]: can you rephrase that?

Comment (bitbauer) [inline lib/vagrant-vbguest/installers/linux.rb]: Hi James, when the method is called in the code it is referenced by vboxadd_tool(), but it is declared by find_tool()

Comment (cdloh) [inline lib/vagrant-vbguest/installers/linux.rb]: yup apologies that's my fault. I've made that edit locally but for some reason it didn't make it into the commit. I'll update that.

**Comment by cdloh** [comment]
Updated. Do you want me to squash the commits?

File: lib/vagrant-vbguest/installers/linux.rb (hunk 1 above)
Comment (fnordfish) [inline lib/vagrant-vbguest/installers/linux.rb]: Hi, do you think it's safe to assume, that if we found `vboxadd` in any path, that the same path will also include `vboxadd-service`? If possible, I'd like to cache the "too path" and avoid running that inline shell script twice.

**Comment by cdloh** [comment]
@fnordfish thanks for the merge. Did you want me to create another with the ability to cache the tool path?

I did some testing and I couldn't find any OSs where it wasn't in the same location.

I however did notice on Suse that the command didn't exist at all.

**Comment by fnordfish** [comment]
@cdloh would be great. Thanks for your work!
